src/lib/Server/Output/ValueObjectVisitor/Location.php

```diff
@@ -174,4 +184,31 @@ protected function visitLocationAttributes(Visitor $visitor, Generator $generato
 
         $generator->endObjectElement('ContentInfo');
     }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    private function resolveMainLocation(
+        Content\ContentInfo $contentInfo,
+        Content\Location $location
+    ): ?Content\Location {
+        $mainLocationId = $contentInfo->mainLocationId;
+        if ($mainLocationId === null) {
+            $this->logger->warning(
+                sprintf('Main location for content of ID = %d doesn\'t exist.', $contentInfo->id)
+            );
+
+            return null;
```

Comment:
TBH given it's an API-supported use case, even though not very common, I have doubt if it should be logged.
Maybe different channel?